library/modules/bigip_gtm_dns_listener.py

```diff
@@ -0,0 +1,890 @@
+        elif any(x.lower() for x in self._values['enabled_vlans'] if x.lower() in ['all', '*']):
+            result = [fq_name(self.partition, 'all')]
+            self._values['vlans_disabled'] = True
+            self._values['vlans_enabled'] = False
+            return result
+        results = list(set([fq_name(self.partition, x) for x in self._values['enabled_vlans']]))
+        results.sort()
+        return results
+
+    @property
+    def disabled_vlans(self):
+        if self._values['disabled_vlans'] is None:
+            return None
+        elif any(x.lower() for x in self._values['disabled_vlans'] if x.lower() in ['all', '*']):
+            raise F5ModuleError(
+                'You cannot disable all VLANs. You must name them individually.'
+            )
+        results = list(set([fq_name(self.partition, x) for x in self._values['disabled_vlans']]))
+        results.sort()
+        return results
+
+    @property
+    def vlans(self):
+        disabled = self.disabled_vlans
+        if disabled:
+            return self.disabled_vlans
+        return self.enabled_vlans
+
+    @property
+    def irules(self):
+        results = []
+        if self._values['irules'] is None:
+            return None
+        if is_empty_list(self._values['irules']):
+            return ''
+        for irule in self._values['irules']:
+            result = fq_name(self.partition, irule)
+            results.append(result)
+        return results
+
+
+class Changes(Parameters):
+    def to_return(self):
+        result = {}
+        try:
+            for returnable in self.returnables:
+                result[returnable] = getattr(self, returnable)
+            result = self._filter_params(result)
+        except Exception:
+            raise
+        return result
+
+
+class UsableChanges(Changes):
+    @property
+    def vlans(self):
+        if self._values['vlans'] is None:
+            return None
+        elif len(self._values['vlans']) == 0:
+            return []
+        elif any(x for x in self._values['vlans'] if x.lower() in ['/common/all', 'all']):
+            return []
+        return self._values['vlans']
+
+
+class ReportableChanges(Changes):
+    @property
+    def enabled_vlans(self):
+        if self._values['vlans'] is None:
+            return None
+        if len(self._values['vlans']) == 0 and self._values['vlans_disabled'] is True:
+            return 'all'
+        elif len(self._values['vlans']) > 0 and self._values['vlans_enabled'] is True:
+            return self._values['vlans']
+
+    @property
+    def disabled_vlans(self):
+        if self._values['vlans'] is None:
+            return None
+        if len(self._values['vlans']) > 0 and self._values['vlans_disabled'] is True:
+            return self._values['vlans']
+
+    @property
+    def irules(self):
+        if self._values['irules'] is None:
+            return None
+        if len(self._values['irules']) == 0:
+            return []
```

Comment:
if the list is empty just do truth check, empty lists, strings etc, always are False:

`if not self._values['irules']`

Reply:
Right - I have made the change as suggested
This comment is addressed now.